### DISCUSSION TITLE
Deep sleep and wake up by external key - idea to implement ?

### DISCUSSION THREAD

**Discussion by z34pec** [Edzelf/ESP32Radio-V2#116] [Ideas]
Finally I found this web page where it is explained how to make a hardware for deep sleep and wake up of esp32.

[Deep sleep and wake up buttons](https://circuitdigest.com/microcontroller-projects/esp32-active-mode-and-deep-sleep-mode-power-consumption)

![image](https://github.com/Edzelf/ESP32Radio-V2/assets/133197383/92991c53-60c7-4e01-a25d-e016d473b1bc)

I think GPIO16 and 4 are not and 33 are free for this add on. 
Is there a way to use free gpio's for this function ?

**Comment by Edzelf** [comment]
I added a "sleep" command.  So now you can use the webinterface or any other input (IR, pin) to put the radio into deep sleep.  The only way to wake-up the radio is the reset button.

**Comment by z34pec** [comment]
> I added a "sleep" command. So now you can use the webinterface or any other input (IR, pin) to put the radio into deep sleep. The only way to wake-up the radio is the reset button.

pin_shutdownx = 16          # Deemphasis, will be set to LOW 

Is this the new command for shutdown radio ? If not where we can define pin shutdown ?

In above link I send, there is wake up button (   esp_sleep_enable_ext0_wakeup((gpio_num_t)wakeUp_pin, 0); ) definition, and maybe it is better way of wake up than reset button ?

I think there is a lot of other GPIO pins that can be defined for wake up pins: 5, 12,13,14, 22, 32 and 33 (derived and compared from you used pin list in documentation and give pinout scheme for esp32 )

Can you also define this function in your code ?

**Comment by Edzelf** [comment]
pin_shutdown has nothing to do with deep-sleep.  It is for shutting down the amplifier (if supported).
You may add a command in your configuration (Web-interface) like:
`gpio_16 = sleep`
or
`ir_C5F3 = sleep`
For wake-up use the "EN" pin.  This will also initialize other devices, like display and VC1053.

**Reply by z34pec** [reply]
" For wake-up use the "EN" pin. This will also initialize other devices, like display and VC1053."

What size of resistor is preffered to be put between push button and pin that is used for sleep command ?
How to use EN pin ? Like put push button between EN and GND with let say 1K resistor or a greater value ?

Once again, what about circuit diagram like shown above ? What are its pros and cons ?

**Comment by Edzelf** [comment]
There is normally no need to use pull-up resistors.  Most of the input pint will be configured as "INPUT_PULL_UP" by the software.  The "EN" pin also have a PULL-UP resistor, otherwise the board would not run...
The pins 34 to 39 do not have internal pull-ups, so for these pins you need an external puul-up of about 10k ohm.
So connect the button btetween the input pin and ground without any resistors.  For GPIO 34, 35, 36 and 39: add a 10k pull-up between the input pin and 3.3V.
